Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py

```diff
@@ -45,52 +46,67 @@ def update_incident_with_required_keys(incidents, required_keys):
             incident[key] = updated_incident.get(key)
 
 
-def get_incident_val(incident, key):
+def convert_incident_to_hr(incident):
     """
         Get the value from incident dict and convert it in some cases e.g. make id linkable etc.
+        Note: this script change the original incident
 
         :type incident: ``dict``
         :param incident: the incident to get the value from
 
         :type key: ``str``
         :param key: the key in dict
 
-        :rtype: ``str``
-        :return the value form dict
+        :rtype: ``None``
+        :return None
     """
-    if key == 'status':
-        return STATUS_DICT.get(incident.get(key))
+    converted_incident = copy.deepcopy(incident)
+
+    for key in converted_incident.keys():
+
+        if key == 'status':
+            converted_incident[key] = STATUS_DICT.get(converted_incident.get(key))
+
+        if key == 'id':
+            converted_incident[key] = LINKABLE_ID_FORMAT.format(incident_id=converted_incident.get(key))
 
-    if key == 'id':
-        return LINKABLE_ID_FORMAT.format(incident_id=incident.get(key))
+        converted_incident[key] = converted_incident.get(key.replace('_', ''))
 
-    return incident.get(key.replace('_', ''))
+    return converted_incident
 
 
 def get_campaign_incidents_from_context():
     return demisto.get(demisto.context(), 'EmailCampaign.incidents')
 
 
-def get_incidents_info_md(incidents):
+def get_incidents_info_md(incidents, fields_to_display=None):
     """
         Get the campaign incidents relevant info in MD table
 
         :type incidents: ``list``
         :param incidents: the campaign incidents to collect the info from
+        :type fields_to_display: ``list``
+        :param fields_to_display: list of result headers
 
         :rtype: ``str``
         :return the MD table str
 
     """
+
     if incidents:
-        incidents_info = [
-            {key: get_incident_val(incident, key) for key in INCIDENTS_HEADER} for incident in incidents
-        ]
+        if not fields_to_display:
+            headers = HEADER_SORTED
+        else:
+            headers = [header for header in HEADER_SORTED if fields_to_display]
+            headers += list(set(fields_to_display) - set(HEADER_SORTED))
```

Comment:
add a dev comment about what you are doing here as it looks funny though the idea if I get it is good - adding the rest of the field to display after the preset fields.

Reply:
This section has been changed.